nixos/modules/services/games/asf.nix

```diff
@@ -0,0 +1,200 @@
+      description = "The Web-UI hosted on 127.0.0.1:1242.";
+    };
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.ArchiSteamFarm;
+      description =
+        "Package to use. Should always be the latest version, for security reasons and since this module uses very new features.";
+    };
+
+    dataDir = mkOption {
+      type = types.path;
+      default = "/var/lib/asf";
+      description = "State directory of Archis-Steam-Farm";
+    };
+
+    config = mkOption {
+      type = types.attrs;
+      description = ''
+        The ASF.json file, all the options are documented here: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Configuration#global-config.
+        Do note that AutoRestart is automatically set to false, since nix takes care of updating everything.
+        UpdateChannel is also set to 0 for the same reason.
+        Headless is also set to true, since there is no way to provide input via a systemd service.
+        You should try to keep ASF up to date since upstream does not provide support for anything but hte latest version and you're exposing yourself to all kinds of issues - as is outlined here: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Configuration#updateperiod
+      '';
+      example = {
+        Statistics = false;
+      };
+      default = { };
+    };
+    bots = mkOption {
+      type = types.attrsOf (types.submodule {
+        options = {
+          username = mkOption {
+            type = types.str;
+            description =
+              "Name of the user to log in. Default is attribute name.";
+            default = "";
+          };
+          passwordFile = mkOption {
+            type = types.path;
+            description =
+              "Path to a file containig the password. The file must be readable by the <literal>asf</literal> user/group.";
+          };
+          enabled = mkOption {
+            type = types.bool;
+            default = true;
+            description = "Whether to enable the bot on startup.";
+          };
+          settings = mkOption {
+            type = types.attrs;
+            description =
+              "Additional settings that are documented here: https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Configuration#bot-config";
+            default = { };
+          };
+        };
+      });
+      description = ''
+        Bots name and configuration
+      '';
+      example = {
+        exampleBot = {
+          username = "alice";
+          passwordFile = "/my/super/secret/password";
+          settings = { SteamParentalCode = "1234"; };
+        };
+      };
+      default = { };
+    };
+  };
+
+  config = mkIf cfg.enable {
+
+    users = {
+      users.asf = {
+        home = cfg.dataDir;
+        createHome = true;
+        isSystemUser = true;
+        group = "asf";
+        description = "Archis-Steam-Farm service user";
+      };
+      groups.asf = { };
+    };
+
+    systemd.services = {
+      asf = {
```

Comment:
I notice that the upstream unit only provides `systemd` template services. Can you explain why upstream does that, and why you do not do that here?

Reply:
I have no idea how to use that with nixos, and I haven't seen any other services do that 😅 . I also don't really understand the need for several ASF services running at the same time, since one can already run an infinite amount of bots.

Reply:
I'm not familiar with the software and I have no idea what steam cards even do... so I'll take your word on that 😄